### PR TITLE
[Merged by Bors] - feat: port FieldTheory.SplittingField.Construction

### PR DESCRIPTION
**Pull request by ChrisHughes24**
---
I've started a slightly different approach to this PR. So far I've constructed a `SplittingFieldAux` without worrying about whether the instances satisfy nice definitional equalities. Then the actual `SplittingField` is defined to be a quotient of an `MvPolynomial` ring by the kernel of the obvious map into `SplittingFieldAux`. Because the actual `SplittingField` will be a quotient of a `MvPolynomial` it should have nice instances on it. 

This is just a suggestion, if somebody prefers doing it the old way and reviving the old PR, I won't object to that.
 

- [x] depends on: #5022

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by Ruben-VandeVelde** [comment]
@ChrisHughes24 in case you forgot, there may be fixes in #4339 that can be salvaged

**Comment by ChrisHughes24** [comment]
Maybe we should add a co authored by because I stole code from the previous PR.

**Comment by riccardobrasca** [comment]
Should we wait to fix the mathlib3 problems before merging this? The relevant PR is [#19178](https://github.com/leanprover-community/mathlib/pull/19178).

**Comment by eric-wieser** [comment]
I think we should forward-port [#19179](https://github.com/leanprover-community/mathlib/pull/19179) first.

**Comment by semorrison** [comment]
This PR/issue depends on:
* leanprover-community/mathlib4#5022
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by semorrison** [comment]
This PR/issue depends on:
* ~~leanprover-community/mathlib4#5022~~
By **[Dependent Issues](https://github.com/z0al/dependent-issues)** (🤖). Happy coding!

**Comment by jcommelin** [comment]
Now that Lean 3 is happy with the corresponding refactor, let's get this merged!

File: Mathlib/Data/MvPolynomial/Basic.lean [hunk 1]
```diff
@@ -150,6 +150,14 @@ instance isScalarTower' [CommSemiring R] [CommSemiring S₁] [Algebra R S₁] :
   IsScalarTower.right
 #align mv_polynomial.is_scalar_tower' MvPolynomial.isScalarTower'
 
+--Porting note: new instance
+instance isScalarTower_right [CommSemiring R] [CommSemiring S₁] [DistribSMul R S₁]
+    [IsScalarTower R S₁ S₁] : IsScalarTower R (MvPolynomial σ S₁) (MvPolynomial σ S₁) :=
+  ⟨by
+    rintro x p q
+    dsimp [MvPolynomial] at p q ⊢
+    rw [smul_mul_assoc]⟩
```
Comment (eric-wieser) [inline Mathlib/Data/MvPolynomial/Basic.lean:159]: This is replaced in #5070

**Pull request closed without merging**